apps/server/src/shared/infra/collaborative-storage/strategy/nextcloud.strategy.spec.ts

```diff
@@ -120,4 +125,99 @@ describe('NextCloud Adapter Strategy', () => {
 			).rejects.toThrow(NotFoundException);
 		});
 	});
+	const teamIdMock = 'teamIdMock';
+	describe('Delete Team from Nextcloud', () => {
+		beforeAll(() => {
+			jest
+				.spyOn(strategy.httpService, 'get')
+				// needed for proper mocking
+				// eslint-disable-next-line @typescript-eslint/no-unused-vars
+				.mockImplementation((url: string, config?: AxiosRequestConfig): Observable<AxiosResponse> => {
+					let data = {};
+					if (url.endsWith('cloud/groups?search=teamIdMock')) {
+						data = { ocs: { data: { groups: ['testGroupId'] } } };
+					}
+					if (url.endsWith('groupfolders/folders')) {
+						data = { ocs: { data: { testFolderId: { groups: { testGroupId: 0 } } } } };
+					}
+					const resp: AxiosResponse = {
+						data,
+						status: 0,
+						statusText: '',
+						headers: {},
+						config: {},
+					};
+					return of(resp);
+				});
+			jest
+				.spyOn(strategy.httpService, 'delete')
+				// needed for proper mocking
+				// eslint-disable-next-line @typescript-eslint/no-unused-vars
+				.mockImplementation((url: string, config?: AxiosRequestConfig): Observable<AxiosResponse> => {
+					let data = {};
+					if (url.endsWith('/groups/groupId')) {
+						data = { ocs: { data: {} } };
+					}
+					if (url.endsWith('/groupfolders/folders/folderId')) {
+						data = { ocs: { data: {} } };
+					}
+					const resp: AxiosResponse = {
+						data,
+						status: 200,
+						statusText: 'OK',
+						headers: {},
+						config: {},
+					};
+					return of(resp);
+				});
+		});
+		it('should remove the groups and delete the groupfolder ', async () => {
+			await strategy.deleteGroupfolderAndRemoveGroup(teamIdMock);
+			expect(strategy.httpService.get).toBeCalledTimes(2);
+			expect(strategy.httpService.delete).toBeCalledTimes(2);
+		});
+
+		it('should not find the groupId and throw a NotFoundException', async () => {
+			jest
+				.spyOn(strategy.httpService, 'get')
+				// needed for proper mocking
+				// eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
cand you delete the ununsed vars url and config? That should be possible

Reply:
without them the mock wont work anymore...have tried this myself before